plugin/src/templates/edge/utils.ts

```diff
@@ -43,13 +60,63 @@ export const buildResponse = async ({
   request: Request
   context: Context
 }) => {
+  // They've returned the NetlifyMiddleware directly, so we'll call `next()` for them.
+  if ('originalRequest' in result.response) {
+    result.response = await (result.response as unknown as NetlifyMiddleware).next()
+  }
+  // This means it's a Netlify Next response.
+  if ('dataTransforms' in result.response) {
+    const response = result.response as NetlifyNextResponse
```

Comment:
[sand] I think we can use a type guard here to replace the need for the `dataTransforms in result.response` check and casting of the `result.response` to NetlifyNextResponse in the following line with a single line. Something like

```
function isNetlifyNextResponse(response: NextResponse): response is NetlifyNextResponse {
  return 'dataTransforms' in response
}
```

and using it in the `if` should do the trick